config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_alloydbclusters.alloydb.cnrm.cloud.google.com.yaml

```diff
@@ -231,6 +235,12 @@ spec:
                       If not set, defaults to 14 days.
                     type: integer
                 type: object
+              deletionPolicy:
+                description: |-
+                  Policy to determine if the cluster should be deleted forcefully.
```

Comment:
Ideally deletion would be deletion and this wouldn't be needed.  KCC also want to mirror the underlying GCP APIs, and this isn't technically a field on the resource.

It sounds like the big reason this is needed is that we basically require force to be set for secondaries with instances (?).

Another option is that we automatically set force=true when trying to delete a secondary cluster, but maybe that gets more complicated as the API evolves (e.g. maybe at some point it will be possible to delete a secondary instance?)

Reply:
Yes, we do this because there is no deletion operation for secondary instances. And the only way to delete a secondary instance is by deleting the associated secondary cluster.

We do have the `--force` in the deletion API https://cloud.google.com/sdk/gcloud/reference/alloydb/clusters/delete and the `deletion_policy` set to `FORCE` does the same thing.

Yes, we can automatically set force=true for secondary clusters, but by allowing the user to set the deletion_policy field, it is more clear to the users that their operation will forcefully delete the associated secondary instance as well

Reply:
A couple of thoughts (non-blocking):

1. We should add instructions about how to actually trigger the force deletion because we also has a general `deletion-policy` annotation, and it must be set "correctly" before the `deletionPolicy` option can be properly executed.
2. As this is a cascading deletion, the users will need to be aware of the following facts and clean up their resources properly:
    1. When the parent AlloyDBCluster is force-deleted correctly, the AlloyDBInstance resources should be deleted, otherwise it'll cause reconciliation failures (and potential causing alerts and consuming error budget).
    2. When the secondary AlloyDBInstance is deleted by itself, the only thing happening will be the corresponding GCP resource is no longer controlled by Config Connector.